org.wso2.codequalitymatrices/src/main/java/com/wso2/code/quality/metrics/GithubApiCaller.java

```diff
@@ -0,0 +1,213 @@
+     *
+     * @param commitHashToBeSearched commit hash to be searched for issues
+     * @param githubAccessToken      Github access token for accessing github API
+     * @return String representation of the json response
+     * @throws CodeQualityMetricsException results
+     */
+    public String callSearchIssueApi(String commitHashToBeSearched, String githubAccessToken) throws
+            CodeQualityMetricsException {
+        try {
+            defaultProperties.load(inputStream);
+            String url = defaultProperties.getProperty("searchIssueApiUrl") + commitHashToBeSearched;
+            httpGet = new HttpGet(url);
+            httpGet.addHeader(ACCEPT, defaultProperties.getProperty("searchIssueApiUrlHeader"));
+            httpGet.addHeader(AUTHORIZATION, BEARER + githubAccessToken);
+        } catch (IllegalArgumentException e) {
+            throw new CodeQualityMetricsException("The url provided for accessing the Github Search Issue API is " +
+                    "invalid ", e);
+        } catch (IOException e) {
+            throw new CodeQualityMetricsException("IO exception occurred when loading the inputstream to the " +
+                    "properties object", e);
+        }
+        return ApiUtility.callApi(httpGet);
+    }
+
+    /**
+     * This is used for calling the github graphql API.
+     *
+     * @param graphqlJsonStructure JSON input structure for calling the graphql API
+     * @param githubToken          Github access token for accessing github API
+     * @return String representation of the json response
+     * @throws CodeQualityMetricsException results
+     */
+    public String callGraphqlApi(JSONObject graphqlJsonStructure, String githubToken) throws
+            CodeQualityMetricsException {
+        HttpPost httpPost;
+        try {
+            defaultProperties.load(inputStream);
+            String url = defaultProperties.getProperty("githubGraphqlUrl");
+            httpPost = new HttpPost(url);
+            httpPost.addHeader(AUTHORIZATION, BEARER + githubToken);
+            httpPost.addHeader(ACCEPT, defaultProperties.getProperty("githubGraphqlUrlHeader"));
+            StringEntity entity = new StringEntity(graphqlJsonStructure.toString());
+            httpPost.setEntity(entity);
+        } catch (IllegalArgumentException e) {
+            throw new CodeQualityMetricsException("The url provided for accessing the Github Graphql API is " +
+                    "invalid", e);
+        } catch (UnsupportedEncodingException e) {
+            throw new CodeQualityMetricsException("An error occurred when creating the String entity from Json " +
+                    "Structure", e);
+        } catch (IOException e) {
+            throw new CodeQualityMetricsException("IO exception occurred when loading the inputstream to the " +
+                    "properties object", e);
+        }
+        return ApiUtility.callGraphQlApi(httpPost);
+    }
+}
```

Comment:
why add call to al the methods? can we think of something like search, getHistory etc..